deltacat/compute/compactor/steps/materialize.py

```diff
@@ -185,23 +214,55 @@ def _materialize(compacted_tables: List[pa.Table]) -> MaterializeResult:
                 f" to download delta locator {delta_locator} with entry ID {src_file_idx_np.item()}"
                 f" is: {download_delta_manifest_entry_time}s"
             )
-            mask_pylist = list(repeat(False, len(pa_table)))
+            record_numbers_tpl_copy = copy.deepcopy(record_numbers_tpl)
+            record_numbers_length_list = chain.from_iterable(record_numbers_tpl_copy)
+            record_numbers_length = sum(1 for item in record_numbers_length_list)
             record_numbers = chain.from_iterable(record_numbers_tpl)
-            # TODO(raghumdani): reference the same file URIs while writing the files
-            # instead of copying the data over and creating new files.
-            for record_number in record_numbers:
-                mask_pylist[record_number] = True
-            mask = pa.array(mask_pylist)
-            pa_table = pa_table.filter(mask)
-            record_batch_tables.append(pa_table)
-            if record_batch_tables.has_batches():
-                batched_tables = record_batch_tables.evict()
-                materialized_results.append(_materialize(batched_tables))
+            if (
+                record_numbers_length == len(pa_table)
+                and src_file_partition_locator != source_partition_locator
```

Comment:
I would instead ensure that `src_file_partition_locator` is always equal to `round_completion_info.compacted_delta_locator.partition_locator` as it is by design.

Reply:
My understanding is that `src_file_partition_locator` will equal to `source_partition_locator` if this file is coming from new deltas from source table?

Reply:
Yes, unless `src_partition_locator` is equal to destination, we shouldn't be caching deltas as they are completely different streams.

Reply:
Could you explain more on what you mean by src_file_partition_locator always equal to round_completion_info.compacted_delta_locator.partition_locator? There will be two types of delta: one from previous compacted table, one from new uncompacted table. The partition locator is different here right?